components/event-bus/internal/knative/subscription/controller/eventactivation/controller.go

```diff
@@ -1,47 +1,47 @@
 package eventactivation
 
 import (
-	eventactivationv1alpha1 "github.com/kyma-project/kyma/components/event-bus/internal/ea/apis/applicationconnector.kyma-project.io/v1alpha1"
+	"context"
+
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"knative.dev/eventing/pkg/reconciler"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+
+	eventbusscheme "github.com/kyma-project/kyma/components/event-bus/client/generated/clientset/internalclientset/scheme"
+	eventbusclient "github.com/kyma-project/kyma/components/event-bus/client/generated/injection/client"
+	eventactivationinformersv1alpha1 "github.com/kyma-project/kyma/components/event-bus/client/generated/injection/informers/applicationconnector/v1alpha1/eventactivation"
 	"github.com/kyma-project/kyma/components/event-bus/internal/knative/util"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/handler"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logf.Log.WithName("eventactivation-controller")
-
 const (
+	// reconcilerName is the name of the reconciler
+	reconcilerName = "EventActivations"
+
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
 	controllerAgentName = "eventactivation-controller"
 )
 
-// ProvideController instantiates a reconciler which reconciles EventActivations.
-func ProvideController(mgr manager.Manager) error {
-
-	var err error
+func init() {
+	// Add sources types to the default Kubernetes Scheme so Events can be
+	// logged for sources types.
+	eventbusscheme.AddToScheme(scheme.Scheme)
+}
 
-	// Setup a new controller to Reconcile EventActivation.
-	r := &reconciler{
-		recorder: mgr.GetRecorder(controllerAgentName),
-		time:     util.NewDefaultCurrentTime(),
-	}
-	c, err := controller.New(controllerAgentName, mgr, controller.Options{
-		Reconciler: r,
-	})
-	if err != nil {
-		log.Error(err, "Unable to create controller")
-		return err
-	}
+// NewController returns a new controller that reconciles EventActivation objects.
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	eventActivationInformer := eventactivationinformersv1alpha1.Get(ctx)
 
-	// Watch EventActivations.
-	err = c.Watch(&source.Kind{
-		Type: &eventactivationv1alpha1.EventActivation{},
-	}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		log.Error(err, "Unable to watch EventActivation")
-		return err
+	r := &Reconciler{
+		Base:                       reconciler.NewBase(ctx, controllerAgentName, cmw),
+		eventActivationLister:      eventActivationInformer.Lister(),
+		applicationconnectorClient: eventbusclient.Get(ctx).ApplicationconnectorV1alpha1(),
+		kymaEventingClient:         eventbusclient.Get(ctx).EventingV1alpha1(),
+		time:                       util.NewDefaultCurrentTime(),
 	}
+	impl := controller.NewImpl(r, r.Logger, reconcilerName)
 
```

Comment:
We need to watch for changes to EventActivation objects, otherwise the reconciliation won't get triggered.

```golang
// set event handlers
eventActivationInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
```

Reply:
Can you please elaborate?

Reply:
Whenever an `EventActivation` object is Created/Updated/Deleted in the informer's cache (`controller.HandleAll`), you need to act on that event and trigger the reconciliation (`impl.Enqueue`).

Reply:
Thanks!